### ISSUE TITLE
TransactionData is broken in 2.2.3 and 2.3-M02. RelationshipTypes of changed and deleted relationships are incorrect.

### ISSUE THREAD

**Issue by bachmanm** [neo4j/neo4j#4910]
```
@Test
public void demonstrateBug() {
    GraphDatabaseService db = new TestGraphDatabaseFactory().newImpermanentDatabase();

    db.execute("CREATE (m:Person {name:'Michal'})-[:FRIEND_OF]->(l:Person {name:'Luanne'}), " +
            "(m)-[:LIVES_IN {since:2010}]->(c:City {name:'London'})");


    final Set<String> changedRelationships = new HashSet<>();

    db.registerTransactionEventHandler(new TransactionEventHandler.Adapter<Void>() {
        @Override
        public Void beforeCommit(TransactionData data) throws Exception {
            for (PropertyEntry<Relationship> entry : data.assignedRelationshipProperties()) {
                changedRelationships.add(entry.entity().getType().name());
            }

            return null;
        }
    });

    db.execute("MATCH (m:Person {name:'Michal'})-[r:LIVES_IN]->() SET r.since = 2009");

    assertEquals(1, changedRelationships.size());
    assertTrue(changedRelationships.contains("LIVES_IN")); //fails, contains FRIEND_OF!!!
}

@Test
public void demonstrateBug2() {
    final GraphDatabaseService db = new TestGraphDatabaseFactory().newImpermanentDatabase();

    db.execute("CREATE (m:Person {name:'Michal'})-[:FRIEND_OF]->(l:Person {name:'Luanne'}), " +
            "(m)-[:LIVES_IN {since:2010}]->(c:City {name:'London'})");


    final Set<String> deletedRelationships = new HashSet<>();

    db.registerTransactionEventHandler(new TransactionEventHandler.Adapter<Void>() {
        @Override
        public Void beforeCommit(TransactionData data) throws Exception {
            for (Relationship r : data.deletedRelationships()) {
                deletedRelationships.add(r.getType().name());
            }

            return null;
        }
    });

    db.execute("MATCH (m:Person {name:'Michal'})-[r:LIVES_IN]->() DELETE r");

    assertEquals(1, deletedRelationships.size());
    assertTrue(deletedRelationships.contains("LIVES_IN")); //fails, contains FRIEND_OF!!!
}
```

**Comment by cleishm** [comment]
Hi @bachmanm,

Thanks for raising this. I'll triage it to our team.

Cheers,
Chris

**Comment by tinwelint** [comment]
Closing since this has been fixed by #4943
